examples/pytorch_imagenet_resnet50.py

```diff
@@ -156,13 +165,20 @@ def train(epoch):
             if args.cuda:
                 data, target = data.cuda(), target.cuda()
             optimizer.zero_grad()
-            output = model(data)
-            loss = F.cross_entropy(output, target)
-            loss.backward()
+            # Split batch_size * batches_per_allreduce batch into sub-batches
+            # of size batch_size
+            for i in range(0, actual_batch_size, args.batch_size):
+                data_step = data[i:i + args.batch_size]
+                target_step = target[i:i + args.batch_size]
+                output = model(data_step)
+                train_accuracy.update(accuracy(output, target_step))
+                loss = F.cross_entropy(output, target_step)
+                # Average gradients by the total number of sub-batches
+                loss = loss / args.batches_per_allreduce
+                train_loss.update(loss.item())
```

Comment:
Since we're now always updating loss (which averages instead of summing), we should not need this division by # of batches?

Reply:
Actually, at the end, stats should be averaged by `allreduce_batch_size``, so I think we should multiply instead by ``batch_size`` and then divide by ``allreduce_batch_size``